_posts/2020-08-03-the-mystery-of-the-empty-mail-attribute-in-aad.md

```diff
@@ -0,0 +1,174 @@
+---
+layout: post
+title: The mystery of the empty mail attribute in AAD
+description: A tale of an issue we recently encountered that prevented users from registering for SSPR and MFA
+---
+
+# Introduction
+
+At [Telstra Purple](https://purple.telstra.com), we recently launched a new internal Office 365 tenant.
+The main goal was to give employees from different acquisitions made by Telstra through the years a single roof, which would make collaboration easier.
+
+In early June, we onboarded the first 300 users.
+Unfortunately, we hit an issue that affected a high number of users, which prevented them from getting control of their new account.
+
+In this post, we'll go through how we designed the onboarding process, what went wrong, and how we fixed the issue.
+
+## The onboarding process
+
+Let's first discuss what a working onboarding process looks like.
+
+There are two main steps an employee needs to take:
+
+1. Go through the self-service password reset (SSPR) flow to pick a password;
+1. Register for SSPR and multi-factor authentication (MFA).
+
+In our case, the impacted users faced an issue during the second phase of the account claiming process.
+Before we dive in the issue, let's expand on the strategy we put in place.
+
+### Self-service password reset
+
+The main thing is that we don't want to handle employees' passwords, nor do we want to communicate it to them via email, even if it's possible to force them to update it the first time they use it to log in.
+Instead, we want users to pick a password themselves from the get-go.
+
+To achieve this, we leverage the [self-service password reset capability](https://docs.microsoft.com/en-us/azure/active-directory/authentication/tutorial-enable-sspr) of Azure Active Directory.
+It not only allows employees to pick a password themselves when they get onboarded, it also provides them with a way to reset their password if they forget it or get locked out of their account.
+Less helpdesk involvement means it's a win-win situation for everyone.
+
+Our automation pre-seeds 2 pieces of information from our HR system to ensure that only an employee can reset the password for their account:
```

Comment:
```suggestion
Our automation precedes 2 pieces of information from our HR system to ensure that only an employee can reset the password for their account:
```

Reply:
I don't think that suggestion is correct, maybe my intent is not clear.
I'm not even sure "to seed" is a correct verb 😛

I want to convey the fact that we add some personal information to everyone's account through automation (mobile number and personal email address).
This information is then used during the SSPR flow to verify that only this person can reset the password for this account.

I understand "to precede" as "something that happened before" which doesn't quite match what I mean.
I'll think about a good replacement.

Reply:
I found that official docs on this use "pre-populate", so I'm going to use this as well.

See <https://docs.microsoft.com/en-us/azure/active-directory/authentication/howto-sspr-authenticationdata>